api/machines.go

```diff
@@ -64,6 +83,59 @@ func (mr *machinesResource) ServeHTTP(rw http.ResponseWriter, req *http.Request)
 	sendResponse(rw, http.StatusOK, page)
 }
 
+func (mr *machinesResource) patch(rw http.ResponseWriter, req *http.Request) {
+	ops := make([]machineMetadataOp, 0)
+
+	dec := json.NewDecoder(req.Body)
+
+	if err := dec.Decode(&ops); err != nil {
+		sendError(rw, http.StatusInternalServerError, err)
+		return
+	}
+
+	// Validate all operations before applying
+	for _, op := range ops {
+		if op.Op != "add" && op.Op != "remove" && op.Op != "replace" {
+			sendError(rw, http.StatusBadRequest, fmt.Errorf("Bad op: %s. Expected add/remove/replace", op.Op))
+			return
+		}
+
+		s := metadataPathRegex.FindStringSubmatch(op.Path)
+		if s == nil {
+			sendError(rw, http.StatusBadRequest, fmt.Errorf("Bad path: %s. Expected {machine id}/metadata/{key}. Key must only contain [a-zA-Z0-9_.-]", op.Path))
```

Comment:
This error message is communicating way too much:
1. error messages should always start with a lowercase character
2. don't write multiple sentences
3. don't mirror back user input